src/stratigraphy/groundwater/groundwater_extraction.py

```diff
@@ -269,6 +288,184 @@ def get_groundwater_info_from_lines(self, lines: list[TextLine], page: int) -> G
         else:
             raise ValueError("Could not extract all required information from the lines provided.")
 
+    def load_templates(self) -> list[np.ndarray]:
+        """Load the templates for the groundwater information.
+
+        Returns:
+            list[np.ndarray]: the loaded templates
+        """
+        templates = []
+        template_dir = os.path.join(os.path.dirname(__file__), "assets")
+        for template in os.listdir(template_dir):
+            if template.endswith(".npy"):
+                templates.append(np.load(os.path.join(template_dir, template)))
+        return templates
+
+    def get_groundwater_from_illustration(
+        self, lines: list[TextLine], page_number: int, terrain_elevation: Elevation | None
+    ) -> list[GroundwaterInformationOnPage]:
+        """Extracts the groundwater information from an illustration.
+
+        Args:
+            lines (list[TextLine]): the lines of text to extract the groundwater information from
+            page_number (int): the page number (1-based) of the PDF document
+            terrain_elevation (Elevation | None): The elevation of the terrain.
+
+        Returns:
+            list[GroundwaterInformationOnPage]: the extracted groundwater information
+        """
+        extracted_groundwater_list = []
+        confidence_list = []
+
+        # convert the doc to an image
+        page = self.doc.load_page(page_number - 1)
+        filename = Path(self.doc.name).stem
+        png_filename = f"{filename}-{page_number + 1}.png"
+        png_path = f"/tmp/{png_filename}"  # Local path to save the PNG
+        fitz.utils.get_pixmap(page, matrix=fitz.Matrix(2, 2), clip=page.rect).save(png_path)
+
+        # load the image
+        img = ski.io.imread(png_path)
+        N_BEST_MATCHES = 5
+        TEMPLATE_MATCH_THRESHOLD = 0.66
+
+        # extract the groundwater information from the image
+        for template in self.load_templates():
+            # Compute the match of the template and the image (correlation coef)
+            result = ski.feature.match_template(img, template)
+
+            for _ in range(N_BEST_MATCHES):
+                ij = np.unravel_index(np.argmax(result), result.shape)
+                confidence = np.max(result)  # TODO - use confidence to filter out bad matches
+                if confidence < TEMPLATE_MATCH_THRESHOLD:
+                    # skip this template if the confidence is too low to avoid false positives
+                    continue
+                top_left = (ij[1], ij[0])
+                illustration_rect = fitz.Rect(
+                    top_left[0], top_left[1], top_left[0] + template.shape[1], top_left[1] + template.shape[0]
+                )
+
+                # remove the matched area from the result to avoid finding the same area again
```

Comment:
```suggestion
                # remove the matched area from the template matching result to avoid finding the same area again for the same template
```